File: sprint4/By languages/JS/lesson10code1.js
function naiveSolution(A, x) {
  const n = x.length;
  const triples = [];
  for (let i1 = 0; i1 < n; i1++) {
    for (let i2 = i1 + 1; i2 < n; i2++) {
      for (let i3 = i2 + 1; i3 < n; i3++) {
        if (x[i1] + x[i2] + x[i3] === A) {
          triples.push([x[i1], x[i2], x[i3]]);
        }
      }
    }
  }
  return triples;
}

File: sprint4/By languages/Java/lesson10code1.java
public List<Triple<Integer, Integer, Integer>> naiveSolution(int A, List<Integer> x) {
  int n = x.size();
  List<Triple<Integer, Integer, Integer>> triples = new ArrayList<>();
  for (int i1 = 0; i1 < n; i1++) {
    for (int i2 = i1 + 1; i2 < n; i2++) {
      for (int i3 = i2 + 1; i3 < n; i3++) {
        if (x.get(i1) + x.get(i2) + x.get(i3) == A) {
          triples.add(new Triple<>(x.get(i1), x.get(i2), x.get(i3)));
        }
      }
    }
  }
  return triples;
}
